Problem2/Problem2.js

```diff
@@ -0,0 +1,54 @@
+function mentionStatistics(members, messages) {
+    // Initialize an object to store mention counts for each member
+    const mentionCount = {};
+    members.forEach(member => {
+        mentionCount[member] = 0;
+    });
+
+    // Regular expression to find mentions
+    const mentionRegex = /@id\d{1,3}/g;
+
+    // Process each message
+    messages.forEach(message => {
+        // Extract all mentions in the current message
+        const matches = message.match(mentionRegex) || [];
+        
+        // Use a Set to track unique mentions in the message
+        const uniqueMentions = new Set();
```

Comment:
The Set() structure is a tree.
So, It can be performed all actions like pop, push , delete, update by O(1).